planning/planning_scene_monitor/src/planning_scene_monitor.cpp

```diff
@@ -808,6 +820,41 @@ void planning_scene_monitor::PlanningSceneMonitor::currentWorldObjectUpdateCallb
     }
 }
 
+void planning_scene_monitor::PlanningSceneMonitor::syncSceneUpdates(const ros::Time &t)
+{
+  if (t.isZero())
+    return;
+
+  ROS_DEBUG_STREAM_NAMED("PSM", "sync to: " << fmod(t.toSec(), 10.));
+  enforce_next_state_update_ = true;  // enforce next state update to trigger without throttling
+
+  // Robot state updates in the scene are only triggered by the state monitor on changes of the state.
+  // Hence, last_state_update_time_ might be much older than current_state_monitor_ (when robot didn't moved for a while).
+  boost::shared_lock<boost::shared_mutex> lock(scene_update_mutex_);
+  ros::Time last_robot_update = current_state_monitor_ ? current_state_monitor_->getCurrentStateTime() : ros::Time();
+  while (current_state_monitor_ &&                     // sanity check
+         last_robot_update < t &&                      // wait for recent state update
+         (t - last_robot_motion_time_).toSec() < 1.0)  // but only if robot moved in last second
+  {
+    ROS_DEBUG_STREAM_NAMED("PSM", "robot monitor update " << (t-robot).toSec() << " ago"
+                           << " robot state update " << (t-last_robot_motion_time_).toSec() << "ago");
+    new_scene_update_condition_.wait_for(lock, boost::chrono::milliseconds(50));
```

Comment:
Technically it doesn't really make sense to wait using the scene update condition variable,
because we are only interested in joint state updates here.
But then it probably doesn't make much of a difference..